src/demo-app/overlay/overlay-demo.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, provide, ElementRef, ViewChildren, QueryList} from 'angular2/core';
+import {Overlay, OVERLAY_CONTAINER_TOKEN} from '../../core/overlay/overlay';
+import {ComponentPortal, Portal} from '../../core/portal/portal';
+import {BrowserDomAdapter} from '../../core/platform/browser/browser_adapter';
+import {TemplatePortalDirective} from '../../core/portal/portal-directives';
+
+
+@Component({
+  selector: 'overlay-demo',
+  templateUrl: 'demo-app/overlay/overlay-demo.html',
+  styleUrls: ['demo-app/overlay/overlay-demo.css'],
+  directives: [TemplatePortalDirective],
+  providers: [
+    Overlay,
+    provide(OVERLAY_CONTAINER_TOKEN, {useValue: document.body})
+  ]
+})
+export class OverlayDemo {
+  @ViewChildren(TemplatePortalDirective) templatePortals: QueryList<Portal<any>>;
+
+  constructor(public overlay: Overlay, public elementRef: ElementRef) {
+    BrowserDomAdapter.makeCurrent();
```

Comment:
This isn't actually the `BrowserAdapter` from ng2, it's here in `src/core/platform` copied and pasted from `angular/angular`. It's only necessary for transpiling to Dart right now and will eventually be dropped when https://github.com/angular/angular/issues/7561 is done.